### PR TITLE
Enable access to group name

### PR DESCRIPTION
**Pull request by ajbozarth**
Adds two functions `tls1_group_id2name` which returns the `tlsname` for a given `group_id` and `SSL_get0_group_name` which surfaces that name to the "user" 

This is a follow up to my previously rejected PR #20318 and the resulting discussion. It is related to #20594 but I am unsure if it addresses that issue at its core.

Since I am new to this project I would appreciate direction on if I went about surfacing the `tlsname` the correct way here, how I should document this new functionality, and how/where to test it.

I am aware I will need to submit a ICLA and a CCLA (If IBM hasn't submitted one already) before this can be merged.

##### Checklist
- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: ssl/t1_lib.c
Comment (paulidale) [inline ssl/t1_lib.c:761]: Out coding style requires a blank line between declarations and code.

**Review by paulidale — CHANGES_REQUESTED** [review]
We require documentation for the new function and ideally a test case.

**Comment by ajbozarth** [comment]
A few updates:
- I've address the style issues after reviewing https://www.openssl.org/policies/technical/coding-style.html
- I've added a doc file based on reading doc files for similar surrounding functions.
- I will submit my ICLA on Friday (I will have access to a printer/scanner Thursday night)
- I've confirmed that IBM has previously submitted a CCLA
- I am in the process of figuring out where/how to test this. I will be adding/updating a test in `test/sslapitest.c`

**Review by paulidale — APPROVED** [review]

File: doc/man3/SSL_get_curve_name.pod
Comment (hlandau) [inline doc/man3/SSL_get_curve_name.pod:26]: What happens if the handshake hasn't been initiated or completed yet? If this returns NULL in that case, it would be good to mention it.

**Review by hlandau — CHANGES_REQUESTED** [review]
One small nit, other than that looks good.

File: doc/man3/SSL_get_curve_name.pod
Comment (ajbozarth) [reply doc/man3/SSL_get_curve_name.pod:26]: Good point, I'm not actually sure if it would return NULL or just break. I will make sure it returns NULL and update the doc to reflect it.

**Comment by ajbozarth** [comment]
I've added some test updates, but am not confident I added enough. Feedback on my tests would be appreciated.

Also as an FYI, I emailed in my ICLA today.

**Review by paulidale — APPROVED** [review]
Tests look good enough.

**Comment by t8m** [comment]
CI failures are relevant

**Comment by ajbozarth** [comment]
I believe I've addressed the CI failures, there was one windows build failure I was not sure about though.

If you could re-trigger the CI for me (since I'm not whitelisted)

**Comment by t8m** [comment]
Please rebase the PR against fresh master branch and drop the merge commit. Merge commits are not allowed.

**Comment by ajbozarth** [comment]
> Please rebase the PR against fresh master branch and drop the merge commit. Merge commits are not allowed.

Since most of my commits were just `wip something` I decided to just patch the diff and force push it. 

Everything should be fine now, though I'm unsure why my CCLA situation is still unresolved (my ICLA was accepted over a week ago).

**Comment by ajbozarth** [comment]
Closed and reopened to grab my CLA

**Review by paulidale — APPROVED** [review]

**Review by t8m — CHANGES_REQUESTED** [review]

File: doc/man3/SSL_get_curve_name.pod
Comment (ajbozarth) [reply doc/man3/SSL_get_curve_name.pod]: I've pushed an update with the name change. I left the function where it is, but if you'd like me to move the two functions to `ssl_lib.c` I can do that.

Comment (ajbozarth) [reply doc/man3/SSL_get_curve_name.pod]: also if you could double check the doc wording, I just replaced the word "curve" with "group" but am unsure if I need to reword the entire sentence instead for the correct meaning

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@hlandau @paulidale please reconfirm

**Comment by ajbozarth** [comment]
FYI I saw a merge conflict message so I rebased on latest master and force pushed it

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by Tarnum-tst** [comment]
'''
SSL_get_curve_name                      ?	3_2_0	EXIST::FUNCTION:
'''
should be removed from util/libssl.num

**Comment by paulidale** [comment]
Needs the fix up before merge.  Approved anew once that's done.

**Comment by ajbozarth** [comment]
just pushed the requested fix

**Comment by t8m** [comment]
Squashed commits and merged to master branch. Thank you for your contribution.

**Comment by ajbozarth** [comment]
Follow up Question:
Is there any way to know when this will be part of a release? From my research it would assume it needs to wait for 3.2, which doesn't seem to have an estimate yet afaik.
I'm working on some upstream usage of the new function and would prefer to have a timeline

**Comment by paulidale** [comment]
No release date for 3.2 has been set at this stage.
I'd expect late this year, but not promises.

**Pull request closed without merging**
